src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/TransferCaseCallbackHandler.java

```diff
@@ -0,0 +1,120 @@
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.hmcts.cmc.ccd.domain.CaseEvent.TRANSFER_CASE;
+import static uk.gov.hmcts.cmc.claimstore.services.ccd.Role.CASEWORKER;
+
+@Service
+@ConditionalOnProperty({"feature_toggles.ctsc_enabled"})
+public class TransferCaseCallbackHandler extends CallbackHandler {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private static final List<Role> ROLES = Collections.singletonList(CASEWORKER);
+    private static final List<CaseEvent> EVENTS = ImmutableList.of(TRANSFER_CASE);
+    private final BulkPrintTransferService bulkPrintTransferService;
+    private final CaseDetailsConverter caseDetailsConverter;
+
+    @Autowired
+    public TransferCaseCallbackHandler(
+        BulkPrintTransferService bulkPrintTransferService,
+        CaseDetailsConverter caseDetailsConverter
+    ) {
+        this.bulkPrintTransferService = bulkPrintTransferService;
+        this.caseDetailsConverter = caseDetailsConverter;
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return ImmutableMap.of(
+            CallbackType.ABOUT_TO_START, this::prepopulateData,
+            CallbackType.ABOUT_TO_SUBMIT, this::performBulkPrintTransfer
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    @Override
+    public List<Role> getSupportedRoles() {
+        return ROLES;
+    }
+
+    private CallbackResponse prepopulateData(CallbackParams callbackParams) {
+
+     return AboutToStartOrSubmitCallbackResponse
+            .builder()
+            .data(ImmutableMap.of(
+                "dateOfTransfer", "TODAYS_DATE", // TODO
+                "reasonForTransfer",null,
+                "nameOfTransferCourt", null,
+                "addressOfTransferCourt", null
+            ))
+            .build();
+    }
+
+    private CallbackResponse performBulkPrintTransfer(CallbackParams callbackParams) {
+
+        CCDCase ccdCase = caseDetailsConverter.extractCCDCase(callbackParams.getRequest().getCaseDetails());
+
+        try {
+
+            var noticeOfTransferLetter = bulkPrintTransferService.generateNoticeOfTransferLetter(ccdCase);
+
+            bulkPrintTransferService.addNoticeOfTransferToCaseRecordToCaseDocuments(ccdCase, noticeOfTransferLetter);
+
+            if (!isDefendantLinked(ccdCase)) {
+                var transferNoticeLetter = bulkPrintTransferService.generateTransferNoticeLetter(ccdCase);
+
+                bulkPrintTransferService.addTransferLetterNoticeToCaseDocuments(ccdCase, transferNoticeLetter);
+            }
+
+            bulkPrintTransferService.sendCaseDocumentsToBulkPrint(ccdCase);
+
+            bulkPrintTransferService.sendClaimUpdatedEmailToClaimant(ccdCase);
+
+            if (isDefendantLinked(ccdCase)) {
+                bulkPrintTransferService.sendClaimUpdatedEmailToDefendant(ccdCase);
+            }
```

Comment:
This business logic probably should just be in `BulkPrintTransferService`

Reply:
I might change this later. I wanted a public method for each step so that a unit test could be written for each with only mocks required for each individual test.